docs/SETUP_TLS_CERTIFICATES.md

```diff
@@ -0,0 +1,232 @@
+# Setting up TLS certificates for secure plugin communication in Snap
+
+<!-- TOC -->
+
+- [Setting up TLS certificates for secure plugin communication in Snap](#setting-up-tls-certificates-for-secure-plugin-communication-in-snap)
+    - [Overview](#overview)
+        - [Remarks](#remarks)
+    - [Obtaining self-signed certificates](#obtaining-self-signed-certificates)
+        - [Remarks](#remarks-1)
+        - [Procedure](#procedure)
+    - [Obtaining legitimate certificate from a public CA Let's Encrypt](#obtaining-legitimate-certificate-from-a-public-ca-lets-encrypt)
+        - [Prerequisites](#prerequisites)
+        - [Remarks](#remarks-2)
+        - [Procedure](#procedure-1)
+    - [Installing certificates in system (Ubuntu, CentOS)](#installing-certificates-in-system-ubuntu-centos)
+        - [Installing root certificate in Ubuntu](#installing-root-certificate-in-ubuntu)
+        - [Installing root certificate in Centos](#installing-root-certificate-in-centos)
+    - [Testing TLS-secured plugin communication in local setup](#testing-tls-secured-plugin-communication-in-local-setup)
+        - [Starting a telemetry workflow:](#starting-a-telemetry-workflow)
+
+<!-- /TOC -->
+
+
+## Overview
+
+This document provides instructions on how to obtain certficates to evaluate TLS-secured plugin communication in Snap.
+
+Following routines are described:
+- obtaining self-signed certificates for evaluation in local test environment,
+- obtaining legitimate certificate from a certain CA authority,
+- installing root certificate in OS for validating endpoints' certificates,
+- starting telemetry workflow with secured plugin communication.
+
+### Remarks
+- This document can't be considered a definite guide to obtaining certificates. It aims only as guiding user in acquiring a certificate.
+- This guide doesn't indicate how to handle certificate revocation as that service is not yet supported in Snap.
```

Comment:
Same, maybe: 

> Note that this guide doesn't cover how to handle certificate revocation as that service is not yet supported in Snap

☝️  is there an issue for that that someone should follow or do we want to suggest people open an issue if they need this feature? Either way, it's helpful when making a statement of what Snap does not do to point people to where they can either (A) offer to help or (B) track the/comment on the need.